app/controllers/merchants_controller.rb

```diff
@@ -72,8 +74,12 @@ def toggle_item
   private
 
   def item_params
-    thing = params[:item][:thumbnail].original_filename
-    params[:item][:thumbnail] = thing
-    params.require(:item).permit(:name, :description, :price, :thumbnail, :inventory)
+    if params[:item][:thumbnail]
+      thing = params[:item][:thumbnail].original_filename
+      params[:item][:thumbnail] = thing
+      params.require(:item).permit(:name, :description, :price, :thumbnail, :inventory)
+    else 
+      params.require(:item).permit(:name, :description, :thumbnail, :price, :inventory)
+    end
   end
```

Comment:
it looks like 

```
params.require(:item).permit(:name, :description, :price, :thumbnail, :inventory)
```

happens no matter what. I would take out of conditional

Reply:
nah I change the params in the first if